src/code/z_message_PAL.c

```diff
@@ -3259,8 +3259,8 @@ void Message_Update(GlobalContext* globalCtx) {
             } else {
                 msgCtx->textboxEndType = TEXTBOX_ENDTYPE_DEFAULT;
             }
-            if ((s32)(gSaveContext.inventory.questItems & 0xF0000000) == 0x40000000) {
-                gSaveContext.inventory.questItems ^= 0x40000000;
+            if ((s32)(gSaveContext.inventory.questItems & 0xF0000000) == (1 << QUEST_HEART_CONTAINER)) {
```

Comment:
I feel like `(1 << QUEST_HEART_CONTAINER)` would be more legible as
```suggestion
            if ((s32)(gSaveContext.inventory.questItems & 0xF0000000) == (4 << QUEST_HEART_PIECE)) {
```

also maybe `QUEST_HEART_PIECE` should be plural, or `QUEST_HEART_PIECE_COUNT`?

Reply:
Oh nice, I actually like that idea more. I think of the two suggestions, I like `QUEST_HEART_PIECE_COUNT` a tiny bit better since it's more specific